### PR TITLE
TBS LUCID H7

### PR DESCRIPTION
**Pull request by andyp1per**
TBS LUCID H7 board

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@andyp1per images?

**Comment by andyp1per** [comment]
> @andyp1per images?

Waiting on TBS for some answers and info

**Comment by Hwurzburg** [comment]
need gh connectors pinouts.....bottom photo too blurry to read pin labels

**Comment by Hwurzburg** [comment]
great! will review in the next day or two...thanks

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+The TBS LUCID H7 is a flight controller produced by [TBS](https://www.team-blacksheep.com/).
+
+## Features
+
+ - MCU - STM32H743 32-bit processor running at 480 MHz
+ - IMU - Dual ICM42688
+ - Barometer - DPS310
+ - OSD - AT7456E
+ - microSD card slot
+ - 7x UARTs
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:12]:
```suggestion
 - 7x UARTs
 - CAN Port
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ - 13x PWM Outputs (12 Motor Output, 1 LED)
+ - Battery input voltage: 2S-6S
+ - BEC 3.3V 0.5A
+ - BEC 5V 3A
+ - BEC 9V 3A for video, gpio controlled
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:17]:
```suggestion
 - BEC 9V 3A for video, gpio controlled, pinned out on HD VTX connector
 - Selectable 5V or VBAT pad, for analog VTX, gpio controlled on/off
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+![TBS LUCID H7 Board Top](Top.png "TBS LUCID H7 Top")
+![TBS LUCID H7 Board Bottom](Bottom.png "TBS LUCID H7 Bottom")
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:30]:
```suggestion
 - SERIAL0 -> USB (MAVLink2)
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+ - SERIAL1 -> UART1 (SBUS in DJI connector, requires SERIAL1_PROTOCOL = 23 to be used)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:31]:
```suggestion
 - SERIAL1 -> UART1 (RX1 is SBUS in HD VTX connector)
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+ - SERIAL2 -> UART2 (GPS, DMA-enabled)
+ - SERIAL3 -> UART3 (DisplayPort, DMA-enabled)
+ - SERIAL4 -> UART4 (Telem1)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:34]:
```suggestion
 - SERIAL4 -> UART4 (MAVLink2,Telem1)
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+ - SERIAL6 -> UART6 (RC Input, DMA-enabled)
+ - SERIAL7 -> UART7 (Telem2, DMA and flow-control enabled)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:36]:
```suggestion
 - SERIAL7 -> UART7 (MAVLink2,Telem2, DMA and flow-control enabled)
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+ - SERIAL8 -> UART8 (ESC Telemetry)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:37]:
```suggestion
 - SERIAL8 -> UART8 (ESC Telemetry, RX8 on ESC connector for telem)
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+## RC Input
+
+RC input is configured by default via the USART6 RX input. It supports all serial RC protocols except PPM.
+
+Note: If the receiver is FPort the receiver must be tied to the USART6 TX pin , RSSI_TYPE set to 3,
+and SERIAL6_OPTIONS must be set to 7 (invert TX/RX, half duplex). For full duplex like CRSF/ELRS use both
+RX6 and TX6 and set RSSI_TYPE also to 3.
+
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:46]:
```suggestion
 If SBUS is used on HD VTX connector (DJI TX), then SERIAL1_PROTOCOl should be set to "23" and SERIAL6_PROTOCOL changed to something else.
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+RC input is configured by default via the USART6 RX input. It supports all serial RC protocols except PPM.
+
+Note: If the receiver is FPort the receiver must be tied to the USART6 TX pin , RSSI_TYPE set to 3,
+and SERIAL6_OPTIONS must be set to 7 (invert TX/RX, half duplex). For full duplex like CRSF/ELRS use both
+RX6 and TX6 and set RSSI_TYPE also to 3.
+ 
+## FrSky Telemetry
+ 
+FrSky Telemetry is supported using an unused UART, such as the T3 pin (UART3 transmit).
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:49]:
```suggestion
FrSky Telemetry is supported using an unused UART, such as the T1 pin (UART11 transmit).
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+You need to set the following parameters to enable support for FrSky S.PORT:
+ 
+  - SERIAL3_PROTOCOL 10
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:52]:
```suggestion
  - SERIAL1_PROTOCOL 10
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+  - SERIAL3_OPTIONS 7
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:53]:
```suggestion
  - SERIAL1_OPTIONS 7
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+LiPo batteries.
+
+The correct battery setting parameters are:
+
+ - BATT_MONITOR 4
+ - BATT_VOLT_PIN 11
+ - BATT_CURR_PIN 13
+ - BATT_VOLT_MULT 11.1
+ - BATT_AMP_PERVLT 40
+
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:91]:
```suggestion
Pads for a second analog battery monitor are provided. To use:

- Set BATT2_MONIOTOR 4
- BATT2VOLT_PIN 18
- BAT2_CURR_PIN 7
- BATT2_VOLT_MULT 11.0
- BATT2_AM_PERVOLT as required
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+## Battery Monitoring
+
+The board has a built-in voltage sensor and external current sensor input. The current
+sensor can read up to 130 Amps. The voltage sensor can handle up to 6S
+LiPo batteries.
+
+The correct battery setting parameters are:
+
+ - BATT_MONITOR 4
+ - BATT_VOLT_PIN 11
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:87]:
```suggestion
 - BATT_VOLT_PIN 10
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+ - BATT_CURR_PIN 13
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:88]:
```suggestion
 - BATT_CURR_PIN 11
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+ - BATT_VOLT_MULT 11.1
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:89]:
```suggestion
 - BATT_VOLT_MULT 11.0
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+ - BATT_AMP_PERVLT 40
+
+## Compass
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:92]:
```suggestion
## Analog RSSI and AIRSPEED inputs
Analog RSSI uses RSSI_PIN 8

Analog Airspeed sensor would use ARSPD_PIN 4

## Compass
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+
+The TBS LUCID H7 does not have a builtin compass, but you can attach an external compass using I2C on the SDA and SCL pads.
+
+## VTX power control
+
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:97]: GPIO 81 controls the VSW pins which can be set to output either VBAT or 5V via a board jumper. Setting this GPIO high removes voltage supply to pins. RELAY2 is configured by default to control this GPIO and is low by default.

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+GPIO 83 controls the VTX BEC output to pins marked "9V". Setting this GPIO low removes voltage supply to pins.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:98]:
```suggestion

GPIO 83 controls the VTX BEC output to pins marked "9V" and is included on the HD VTX connector. Setting this GPIO low removes voltage supply to this pin/pad.
```

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+By default RELAY2 is configured to control this pin and sets the GPIO high.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:99]:
```suggestion
By default RELAY4 is configured to control this pin and sets the GPIO high.
```

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
you should add SD card in the booloader to enable firmware update via SD card

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:12]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 3 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 4 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 5 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 6 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 7 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 8 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:48]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 9 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 10 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 11 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 12 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:95]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 13 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 14 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 15 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 16 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:109]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 19 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md (hunk 17 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:114]: @andyp1per still needs GPIO81 stuff added

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/TBS_LUCID_H7/README.md:114]: fixed

**Review by Hwurzburg — APPROVED** [review]

**Pull request merged**